crates/revm/revm-inspectors/src/stack/maybe_owned.rs

```diff
@@ -85,86 +90,92 @@ where
 
     fn log(
         &mut self,
-        evm_data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         address: &Address,
         topics: &[B256],
         data: &Bytes,
     ) {
         match self {
             MaybeOwnedInspector::Owned(insp) => {
-                return insp.borrow_mut().log(evm_data, address, topics, data)
+                return insp.borrow_mut().log(context, address, topics, data)
             }
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
-    fn step_end(&mut self, interp: &mut Interpreter<'_>, data: &mut EVMData<'_, DB>) {
+    fn step_end(&mut self, interp: &mut Interpreter, context: &mut EvmContext<'_, DB>) {
         match self {
-            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, data),
+            MaybeOwnedInspector::Owned(insp) => insp.borrow_mut().step_end(interp, context),
             MaybeOwnedInspector::Stacked(_) => {}
         }
     }
 
     fn call(
         &mut self,
-        data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         inputs: &mut CallInputs,
-    ) -> (InstructionResult, Gas, Bytes) {
+    ) -> Option<(InterpreterResult, Range<usize>)> {
```

Comment:
this api is kinda weird

Reply:
Yeah I'm not convinced the ranges being passed are correct... Need to read into what it signifies